editoast/src/views/projects.rs

```diff
@@ -262,19 +263,31 @@ impl ProjectPatchForm {
 #[patch("")]
 async fn patch(
     data: Json<ProjectPatchForm>,
-    project: Path<i64>,
+    project_id: Path<i64>,
     db_pool: Data<DbPool>,
 ) -> Result<Json<ProjectWithStudies>> {
     let data = data.into_inner();
-    let project_id = project.into_inner();
-    let project = data.into_project(project_id);
-    if let Some(Some(image)) = project.image {
+    let project_id = project_id.into_inner();
+    if let Some(image) = data.image {
         check_image_content(db_pool.clone(), image).await?;
     }
-    let project = match project.update(db_pool.clone()).await? {
-        Some(project) => project,
-        None => return Err(ProjectError::NotFound { project_id }.into()),
+    let conn = &mut db_pool.get().await?;
+    let old_project =
+        Project::retrieve_or_fail(conn, project_id, || ProjectError::NotFound { project_id })
+            .await?;
+    let image_to_delete = if old_project.image != data.image {
+        old_project.image
+    } else {
+        None
     };
+    let project_changeset: ProjectChangeset = data.into();
```

Comment:
```suggestion
    let project_changeset: Changeset<Project> = data.into();
```

And as you said you can implement `update_and_prune_document` in `Changeset<Project>`.